metadata/metadata.go

```diff
@@ -386,35 +390,34 @@ func (c *Client) WriteGuestAttributes(ctx context.Context, key, value string) er
 }
 
 func (c *Client) do(ctx context.Context, cfg requestConfig) (*http.Response, error) {
-	var (
-		urlTokens []string
-		finalURL  string
-	)
+	finalURL, err := url.Parse(cfg.baseURL)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse url: %+v", err)
+	}
+
+	values := finalURL.Query()
 
 	if cfg.hang {
-		urlTokens = append(urlTokens, "wait_for_change=true")
-		urlTokens = append(urlTokens, "last_etag="+c.etag)
+		values.Add("wait_for_change", "true")
+		values.Add("last_etag", c.etag)
 	}
 
 	if cfg.timeout > 0 {
-		urlTokens = append(urlTokens, fmt.Sprintf("timeout_sec=%d", cfg.timeout))
+		values.Add("timeout_sec", fmt.Sprintf("%d", cfg.timeout))
```

Comment:
isn't `timeout` already an int?

Reply:
Yep, it is. `values.Add()` is `values.Add(string, string)` that's why I'm converting it to string.